### PR TITLE
Repulled ex-new Spice: Mtg

### PR DESCRIPTION
**Pull request by puskin**
Sorry, my fault. I wrongly closed my old pull request for Mtg Spice. This is the new one.
Spice idea was taken from here: https://duck.co/ideas/idea/4614/lookup-a-card-from-magic-the-gathering

### CODE REVIEW

**Comment by mtoledo** [comment]
Is there anything left from this pr or can it be merged. It's actually kind of interesting since I've been watching the world championship videos :)

**Comment by puskin94** [comment]
@mtoledo accordingtomy opinion,it  can be merged "as is" otherwise, just tell me improvements ;)

**Comment by MrChrisW** [comment]
@puskin94 Have you addressed the issue with duplicates?  @moollaza also mentions the use of `api_result[0]` is unneeded.

**Comment by puskin94** [comment]
@MrChrisW I think that duplicates are displayed because there are different versions of the same card ( different years of pubblication, but changes only the "layout" of the card, not the details), [0] is needed to avoid this "problem" , displaying the first that array returns

**Comment by jagtalon** [comment]
@puskin94 alright looks like you made some updates. I'll check it out!

**Comment by puskin94** [comment]
@jagtalon thanks mate!

**Comment by jagtalon** [comment]
Hey, @puskin94! Could you pull in master? It looks like there's a merge conflict.

**Comment by puskin94** [comment]
@jagtalon Can you retry now? I'm not sure I fixed the problem!

**Comment by jagtalon** [comment]
@puskin94 looks good! Testing it out now.

**Comment by jagtalon** [comment]
@puskin94 still kinda weird because there are other IAs in here: 

screen shot 2015-01-19 at 3 41 28 pm

File: lib/DDG/Spice/Mtg.pm [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+name "Mtg Card Search";
+description "mtg <card name> Returns infos on the Magic The Gathering Card";
+primary_example_queries "mtg nullify", "mtg Boros Reckoner";
+
+category "entertainment";
+topics "entertainment", "gaming";
+code_url "https://github.com/puskin94/zeroclickinfo-spice/blob/mtg/lib/DDG/Spice/Mtg.pm";
+attribution github => ["https://github.com/puskin94", "puskin"];
+
+triggers start => 'mtg';
```
Comment (jagtalon) [inline lib/DDG/Spice/Mtg.pm]: Maybe it should trigger with "magic" or "magic the gathering" or "magic cards" as well?

File: share/spice/mtg/mtg.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_mtg = function(api_result){
+
+        if (api_result.error) {
```
Comment (jagtalon) [inline share/spice/mtg/mtg.js]: Check if `api_result` exists before accessing its property.

Comment (jagtalon) [inline share/spice/mtg/mtg.js]: i.e.,

```javascript
if (!api_result || api_result.error) {
    ...
}
```

**Comment by puskin94** [comment]
@jagtalon all those files are old PRs that are finished ( i don't know why ) in this branch

**Comment by puskin94** [comment]
Sorry for the mess, when I started to contribute I know like nothing about git. I'm sure next patches will have a better commit history :-)

@jagtalon I think now it's ok!

**Comment by jagtalon** [comment]
@puskin94 much better!

File: lib/DDG/Spice/Mtg.pm [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+use DDG::Spice;
+
+name "Mtg Card Search";
+description "mtg <card name> Returns infos on the Magic The Gathering Card";
+primary_example_queries "mtg nullify", "mtg Boros Reckoner";
+
+category "entertainment";
+topics "entertainment", "gaming";
+code_url "https://github.com/puskin94/zeroclickinfo-spice/blob/mtg/lib/DDG/Spice/Mtg.pm";
```
Comment (jagtalon) [inline lib/DDG/Spice/Mtg.pm]: Change "https://github.com/puskin94" to "https://github.com/duckduckgo"

File: lib/DDG/Spice/Mtg.pm [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+attribution github => ["https://github.com/puskin94", "puskin"];
+
+triggers start => 'mtg', 'magic', 'magic card', 'magic the gathering';
+spice to => 'http://api.mtgdb.info/cards/$1';
```
Comment (jagtalon) [inline lib/DDG/Spice/Mtg.pm]: What are the terms of the API? Can we use it commercially?

File: share/spice/mtg/mtg.css [hunk 5]
```diff
@@ -0,0 +1,7 @@
+.zci--mtg .cardImg {
```
Comment (jagtalon) [inline share/spice/mtg/mtg.css]: Like @moollaza said, please remove this

File: share/spice/mtg/mtg.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_mtg = function(api_result){
+
+        
+        if (!api_result || api_result.error) {
```
Comment (jagtalon) [inline share/spice/mtg/mtg.js]: Should we check if the array length is longer than 0?

**Comment by jagtalon** [comment]
@puskin94 we're currently checking the layout. We'll get back to you

File: lib/DDG/Spice/Mtg.pm [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+name "Mtg Card Search";
+description "mtg <card name> Returns infos on the Magic The Gathering Card";
+primary_example_queries "mtg nullify", "mtg Boros Reckoner";
+
+category "entertainment";
+topics "entertainment", "gaming";
+code_url "https://github.com/puskin94/zeroclickinfo-spice/blob/mtg/lib/DDG/Spice/Mtg.pm";
+attribution github => ["https://github.com/puskin94", "puskin"];
+
+triggers start => 'mtg', 'magic', 'magic card', 'magic the gathering';
```
Comment (jagtalon) [inline lib/DDG/Spice/Mtg.pm]: Can you add example queries to Mtg.t?

**Comment by puskin94** [comment]
@jagtalon I improved all the codes and added the test file. Looking at https://api.mtgdb.info/ I can't see any TOS about the use of their API. Let me Know!

**Comment by puskin94** [comment]
aaaaaaand added a control for the length ;)

File: share/spice/mtg/content.handlebars [hunk 8]
```diff
@@ -0,0 +1 @@
+<div>{{description}}</div>
```
Comment (jagtalon) [inline share/spice/mtg/content.handlebars]: Looks like this isn't used

File: share/spice/mtg/mtg.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            data: api_result[0],
+            meta: {
+                sourceUrl: "http://www.mtgdb.info/cards/" + api_result[0].id,
+                sourceName: "M:tgDb"
+            },
+            normalize: function(item) {
+                return {
+                    title: item.name,
+                    image:  "http://api.mtgdb.info/content/card_images/" + item.id + ".jpeg",
+                    infoboxData: infoboxData
```
Comment (jagtalon) [inline share/spice/mtg/mtg.js]: Could you add a `url` property in here? You can set it to `"http://www.mtgdb.info/cards/" + item.id`. The output is an `a` tag that points to nothing.

screen shot 2015-01-22 at 2 12 08 pm

**Comment by puskin94** [comment]
@jagtalon good news, a new commit is waiting for your approval ;)

File: lib/DDG/Spice/Mtg.pm [hunk 10]
```diff
@@ -0,0 +1,27 @@
+package DDG::Spice::Mtg;
+
+# mtg <card name> Returns infos on the Magic The Gathering Card
+
+use DDG::Spice;
+
+name "Mtg Card Search";
+description "mtg <card name> Returns infos on the Magic The Gathering Card";
```
Comment (jagtalon) [inline lib/DDG/Spice/Mtg.pm]: Last one! Please change it to just "Returns information about Magic The Gathering Cards"

File: share/spice/mtg/mtg.js [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        };
+
+        var infoboxData = [{
+            heading: 'Card Details',
+        }];
+
+        $.each(infoboxItems, function(key, value){
+            if (api_result[0][key]){
+                infoboxData.push({
+                    url: "http://www.mtgdb.info/cards/" + api_result[0].id,
```
Comment (jagtalon) [inline share/spice/mtg/mtg.js]: Should be in the normalize function instead of in here. Shouldn't have a url here.

File: share/spice/mtg/mtg.js [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+            token: "Token",
+            promo: "Promo"
+        };
+
+        var infoboxData = [{
+            heading: 'Card Details',
+        }];
+
+        $.each(infoboxItems, function(key, value){
+            if (api_result[0][key]){
```
Comment (jagtalon) [inline share/spice/mtg/mtg.js]: @puskin94 what do you think of @moollaza's suggestions

**Comment by puskin94** [comment]
@jagtalon sorry, I fixed the line and the description! I hope now this can be ok!

**Comment by puskin94** [comment]
@jagtalon today i had an exam, please forgive me!

**Comment by jagtalon** [comment]
:+1:

**Pull request merged**